pkg/services/store/kind/dashboard/dashboard.go

```diff
@@ -129,6 +132,14 @@ func readDashboard(stream io.Reader, lookup DatasourceLookup) (*dashboardInfo, e
 		}
 
 		switch l1Field {
+		// k8s metadata wrappers (skip)
+		case "metadata", "kind", "apiVersion":
+			_ = iter.Read()
+
+		// recursively read the spec as dashboard json
+		case "spec":
+			return readDashboardIter(iter, lookup)
```

Comment:
^^^ this is the main change here.  when "spec" exists in the dashboard we treat that as the root element.